types/VueRouter.d.ts

```diff
@@ -0,0 +1,90 @@
+import Vue, { AsyncComponent, ComponentOptions, PluginFunction } from 'vue'
```

Comment:
I put it in a separate file because I want to unravel the circular dependency.

VueRouter depends on RouterView and RouterLink
RouterLink depends on NavigationFailure and Route

But `VueRouter` and `NavigationFailure` and `Route` are defined in`types/router.d.ts`.

Should we allow circular dependency here?